packages/backend-function/src/factory.ts

```diff
@@ -182,10 +185,26 @@ class FunctionFactory implements ConstructFactory<AmplifyFunction> {
     outputStorageStrategy,
     resourceNameValidator,
   }: ConstructFactoryGetInstanceProps): AmplifyFunction => {
+    // TODO this should be conditional on some signal,
+    // for example that any access to data was granted for this function.
     if (!this.generator) {
+      const dataResources = constructContainer
+        .getConstructFactory<AmplifyData>('DataResources')
+        ?.getInstance({
+          constructContainer,
+          outputStorageStrategy,
+          resourceNameValidator,
+        });
+      let modelIntrospectionSchema: string | undefined;
+
+      if (dataResources && dataResources.modelIntrospectionSchema) {
+        modelIntrospectionSchema = dataResources.modelIntrospectionSchema;
+      }
```

Comment:
This can now be consolidated down to:
`const modelIntrospectionSchema = dataResources?.modelIntrospectionSchema`
once the type is added to `AmplifyData`